### ISSUE TITLE
upgrade for compatibility with Laravel 11?

### ISSUE THREAD

**Issue by DeveloperChris** [overtrue/laravel-versionable#68]
Hi

I was wondering if there was a possibilty of upgrading versionable to laravel 11?

Composer outputs these errors when requiring versionable

```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - overtrue/laravel-versionable[5.0.0, ..., 5.1.3] require doctrine/dbal ^3.8 -> satisfiable by doctrine/dbal[3.8.0, 3.8.1, 3.8.2, 3.8.3].
    - overtrue/laravel-versionable[4.5.0, ..., 4.6.0] require doctrine/dbal ^3.0 -> satisfiable by doctrine/dbal[3.0.0, ..., 3.8.3].
    - carbonphp/carbon-doctrine-types 3.2.0 conflicts with doctrine/dbal 3.8.3.
    - carbonphp/carbon-doctrine-types 3.2.0 conflicts with doctrine/dbal 3.7.3.
    - nesbot/carbon 3.2.3 requires carbonphp/carbon-doctrine-types * -> satisfiable by carbonphp/carbon-doctrine-types[3.2.0].
    - symfony/http-foundation v7.0.6 conflicts with doctrine/dbal 3.5.4.
    - symfony/http-foundation v7.0.6 conflicts with doctrine/dbal 3.4.6.
    - symfony/http-foundation v7.0.6 conflicts with doctrine/dbal 3.3.8.
    - symfony/http-foundation v7.0.6 conflicts with doctrine/dbal 3.0.0.
    - laravel/framework v11.2.0 requires nesbot/carbon ^2.72.2|^3.0 -> satisfiable by nesbot/carbon[3.2.3].
    - laravel/framework v11.2.0 requires symfony/http-foundation ^7.0 -> satisfiable by symfony/http-foundation[v7.0.6].
    - overtrue/laravel-versionable[1.0.0, ..., 1.1.5] require php ^7.1.3 -> your php version (8.3.0) does not satisfy that requirement.
    - overtrue/laravel-versionable[2.0.0, ..., 2.3.0] require php ^7.4 -> your php version (8.3.0) does not satisfy that requirement.
    - overtrue/laravel-versionable 2.4.0 requires laravel/framework ^5.8|^6.0|^7.0|^8.0 -> found laravel/framework[v5.8.0, ..., v5.8.38, v6.0.0, ..., v6.20.44, v7.0.0, ..., v7.30.6, v8.0.0, ..., v8.83.27] but it conflicts with your root composer.json require (^11.0).
    - overtrue/laravel-versionable[2.5.0, ..., 2.6.0] require laravel/framework ^5.8|^6.20.14|^7.0|^8.0 -> found laravel/framework[v5.8.0, ..., v5.8.38, v6.20.14, ..., v6.20.44, v7.0.0, ..., v7.30.6, v8.0.0, ..., v8.83.27] but it conflicts with your root composer.json require (^11.0).
    - overtrue/laravel-versionable 2.6.1 requires laravel/framework ^5.8|^6.20.26|^7.0|^8.0 -> found laravel/framework[v5.8.0, ..., v5.8.38, v6.20.26, ..., v6.20.44, v7.0.0, ..., v7.30.6, v8.0.0, ..., v8.83.27] but it conflicts with your root composer.json require (^11.0).
    - overtrue/laravel-versionable[3.0.0, ..., 3.1.4, 4.0.0, ..., 4.2.1] require laravel/framework ^9.0 -> found laravel/framework[v9.0.0, ..., v9.52.16] but it conflicts with your root composer.json require (^11.0).
    - overtrue/laravel-versionable[4.3.0, ..., 4.4.0] require laravel/framework ^9.0|^10.0 -> found laravel/framework[v9.0.0, ..., v9.52.16, v10.0.0, ..., v10.48.4] but it conflicts with your root composer.json require (^11.0).
    - doctrine/dbal[3.1.0, ..., 3.3.5] require doctrine/deprecations ^0.5.3 -> found doctrine/deprecations[v0.5.3] but the package is fixed to 1.1.3 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - laravel/framework is locked to version v11.2.0 and an update of this package was not requested.
    - Root composer.json requires overtrue/laravel-versionable * -> satisfiable by overtrue/laravel-versionable[1.0.0, ..., 1.1.5, 2.0.0, ..., 2.6.1, 3.0.0, ..., 3.1.4, 4.0.0, ..., 4.6.0, 5.0.0, ..., 5.1.3].
```

**Comment by overtrue** [comment]
update with option "-W"?

DeveloperChris ***@***.***>于2024年4月4日 周四12:38写道：

**Comment by DeveloperChris** [comment]
I was able to update doctrine separately  using composer require doctrine/dbal

however I still cannot add versionable I get this error...

```
Problem 1
    - overtrue/laravel-versionable[1.0.0, ..., 1.1.5] require php ^7.1.3 -> your php version (8.3.0) does not satisfy that requirement.
    - overtrue/laravel-versionable[2.0.0, ..., 2.3.0] require php ^7.4 -> your php version (8.3.0) does not satisfy that requirement.
    - overtrue/laravel-versionable 2.4.0 requires laravel/framework ^5.8|^6.0|^7.0|^8.0 -> found laravel/framework[v5.8.0, ..., v5.8.38, v6.0.0, ..., v6.20.44, v7.0.0, ..., v7.30.6, v8.0.0, ..., v8.83.27] but it conflicts with your root composer.json require (^11.0).
    - overtrue/laravel-versionable[2.5.0, ..., 2.6.0] require laravel/framework ^5.8|^6.20.14|^7.0|^8.0 -> found laravel/framework[v5.8.0, ..., v5.8.38, v6.20.14, ..., v6.20.44, v7.0.0, ..., v7.30.6, v8.0.0, ..., v8.83.27] but it conflicts with your root composer.json require (^11.0).
    - overtrue/laravel-versionable 2.6.1 requires laravel/framework ^5.8|^6.20.26|^7.0|^8.0 -> found laravel/framework[v5.8.0, ..., v5.8.38, v6.20.26, ..., v6.20.44, v7.0.0, ..., v7.30.6, v8.0.0, ..., v8.83.27] but it conflicts with your root composer.json require (^11.0).
    - overtrue/laravel-versionable[3.0.0, ..., 3.1.4, 4.0.0, ..., 4.2.1] require laravel/framework ^9.0 -> found laravel/framework[v9.0.0, ..., v9.52.16] but it conflicts with your root composer.json require (^11.0).
    - overtrue/laravel-versionable[4.3.0, ..., 4.4.0] require laravel/framework ^9.0|^10.0 -> found laravel/framework[v9.0.0, ..., v9.52.16, v10.0.0, ..., v10.48.4] but it conflicts with your root composer.json require (^11.0).
    - overtrue/laravel-versionable[4.5.0, ..., 4.6.0] require doctrine/dbal ^3.0 -> found doctrine/dbal[3.0.0, ..., 3.8.3] but it conflicts with your root composer.json require (^4.0).
    - overtrue/laravel-versionable[5.0.0, ..., 5.1.3] require doctrine/dbal ^3.8 -> found doctrine/dbal[3.8.0, 3.8.1, 3.8.2, 3.8.3] but it conflicts with your root composer.json require (^4.0).
    - Root composer.json requires overtrue/laravel-versionable * -> satisfiable by overtrue/laravel-versionable[1.0.0, ..., 1.1.5, 2.0.0, ..., 2.6.1, 3.0.0, ..., 3.1.4, 4.0.0, ..., 4.6.0, 5.0.0, ..., 5.1.3].
```

**Comment by overtrue** [comment]
I've released 5.1.4 plz try again.

**Comment by DeveloperChris** [comment]
That worked

I first removed dbal:4.0

Then tried again and it worked however it was probably due to your changes :) not due to the removal of dbal
